packages/http/src/operations.ts

```diff
@@ -119,10 +119,24 @@ export function getAllRoutes(
 
 export function reportIfNoRoutes(program: Program, routes: HttpOperation[]) {
   if (routes.length === 0) {
-    reportDiagnostic(program, {
-      code: "no-routes",
-      target: program.getGlobalNamespaceType(),
+    const namespaceCounts = new Map<Namespace, number>();
+    navigateProgram(program, {
+      namespace: (namespace) => {
+        namespaceCounts.set(namespace, namespace.operations.size);
```

Comment:
Not sure this intermediate map brings value, could you not just report diagnostic here directly.

Reply:
Good catch. Consolidated.